### ISSUE TITLE
[Bug] SequelizeDatabaseError: there is no unique

### ISSUE THREAD

**Issue by agustinpallares94** [pinecone-io/chatbot-demo#3]
### Is this a new bug?

- [X] I believe this is a new bug
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

I have launched the application with everything needed. I have created  conversation in cockroachdb and when I launch the application I get the following error:

`Error adding entry: SequelizeDatabaseError: there is no unique or exclusion constraint matching the ON CONFLICT specification`

### Expected Behavior

Have the chatbot answer my questions

### Steps To Reproduce

1.Create .env file
2. Launch the application with npm run dev

### Environment

```markdown
- **OS**:
- **Language version**:
- **Pinecone client version**:
```

**Comment by aaronmckeehan** [comment]
There dont seem to be instructions on how to create the Table, but here is what I did, I imagine theres a reason ON Conflict is used, but I changed it to:

`INSERT INTO conversations (user_id, entry, speaker) VALUES (?, ?, ?)`

Here is the create table command I used for Cockroach:
`CREATE TABLE conversations (
        user_id STRING,
        entry STRING,
        speaker STRING,
	created_at TIMESTAMP DEFAULT now()
);`

Since we are running a SELECT on the user ID, and ordering by created time, I don't see why ON CONFLICT is necessary.

**Comment by axelferdinand** [comment]
> There dont seem to be instructions on how to create the Table, but here is what I did, I imagine theres a reason ON CONFLICT is used, but I just changed it to:
> 
> `INSERT INTO conversations (user_id, entry, speaker) VALUES (?, ?, ?)`
> 
> Here is the create table command I used for Cockroach: `CREATE TABLE conversations ( user_id STRING, entry STRING, speaker STRING, created_at TIMESTAMP DEFAULT now() );`
> 
> Since we are running a SELECT on the user ID, and ordering by created time, I don't see why ON CONFLICT is necessary.

Yeah, I was wondering the same thing: How do you actually create the Table, in the first place? To make Pinecone index the pages, you had to rund this URL GET https://localhost:3000/api/crawl?urls=url1,url2&limit=10&indexName=yourIndexName

**Comment by axelferdinand** [comment]
I figured I needed to name the database `conversations`, but now I get this error:

```
[DatabaseError [SequelizeDatabaseError]: relation "conversations" does not exist] {
  parent: [error: relation "conversations" does not exist] {
    length: 110,
    severity: 'ERROR',
    code: '42P01',
    detail: undefined,
    hint: undefined,
    position: undefined,
    internalPosition: undefined,
    internalQuery: undefined,
    where: undefined,
    schema: undefined,
    table: undefined,
    column: undefined,
    dataType: undefined,
    constraint: undefined,
    file: 'errors.go',
    line: '168',
    routine: 'NewUndefinedRelationError',
    sql: "SELECT entry, speaker, created_at FROM conversations WHERE user_id = 'undefined' ORDER By created_at DESC LIMIT 10",
    parameters: undefined
  },
  original: [error: relation "conversations" does not exist] {
    length: 110,
    severity: 'ERROR',
    code: '42P01',
    detail: undefined,
    hint: undefined,
    position: undefined,
    internalPosition: undefined,
    internalQuery: undefined,
    where: undefined,
    schema: undefined,
    table: undefined,
    column: undefined,
    dataType: undefined,
    constraint: undefined,
    file: 'errors.go',
    line: '168',
    routine: 'NewUndefinedRelationError',
    sql: "SELECT entry, speaker, created_at FROM conversations WHERE user_id = 'undefined' ORDER By created_at DESC LIMIT 10",
    parameters: undefined
  },
  sql: "SELECT entry, speaker, created_at FROM conversations WHERE user_id = 'undefined' ORDER By created_at DESC LIMIT 10",
  parameters: {}
  ```

**Comment by luqiang4xenera** [comment]
if you created the conversations table and still show the error [Error adding entry: SequelizeDatabaseError: there is no unique or exclusion constraint matching the ON CONFLICT specification]

you can run sql below
`
ALTER TABLE conversations ADD CONSTRAINT conversations_created_at_unique UNIQUE (created_at);`

**Comment by siddhijain47** [comment]
I am also facing the same error. How to resolve it any idea @axelferdinand @luqiang4xenera . How I can create that table

**Comment by axelferdinand** [comment]
It turned out I simply had to set up and use the right API keys from all the needed services (6 in total, I think) – then it worked as expected, with no code changes :)

**Comment by siddhijain47** [comment]
@axelferdinand Can you share the right way to provide ABLY_API_KEY?

**Comment by axelferdinand** [comment]
<img width="1725" alt="Skjermbilde 2023-07-04 kl  12 26 45" src="https://github.com/pinecone-io/chatbot-demo/assets/10460870/bcc88e98-345d-4f00-a0a9-54cea4c19af7">

**Comment by siddhijain47** [comment]
@axelferdinand I also provided the same as you share but still facing error
![image](https://github.com/pinecone-io/chatbot-demo/assets/62420535/d409f731-5a2e-4c26-8db6-7ae5a5718b98)

**Comment by axelferdinand** [comment]
Try the other API key, the bottom one

**Comment by siddhijain47** [comment]
> Try the other API key, the bottom one

tried from both none work for me

**Comment by siddhijain47** [comment]
> if you created the conversations table and still show the error [Error adding entry: SequelizeDatabaseError: there is no unique or exclusion constraint matching the ON CONFLICT specification]
> 
> you can run sql below ` ALTER TABLE conversations ADD CONSTRAINT conversations_created_at_unique UNIQUE (created_at);`

Is I have to create locally ?  I am expecting when i ran command <npm run dev> it will automatically created

**Comment by iamk3nnyt** [comment]
The schema is not set up and the imports of `Metadata`, `getMatchesFromEmbeddings` are incorrect so you'd have to do code changes in order to make it work but hey @axelferdinand thanks for playing

If you look through the git issues they will help you fill in the missing gaps @siddhijain47
